### PR TITLE
finish add config dir & base ServerURL tasks

### PR DESCRIPTION
**Pull request by mhmdahmedfathi**

### CODE REVIEW

**Comment by mhmdahmedfathi** [comment]
in task description, it isn't specified what we should do after taking configdir so i just keep it until i determine what should it do

File: src/index.ts [hunk 1]
```diff
@@ -99,4 +99,13 @@ void (async () => {
 			return process.exit(ErrorCode.NotFoundRootPath);
 		}
 	}
+
+	if (args['serverUrl']) {
+		const serverUrl = args['serverUrl'];
+		try {
```
Comment (Creatoon) [inline src/index.ts:105]: Why try catch?

Comment (mhmdahmedfathi) [reply src/index.ts:105]: Because i don't know for sure if i have to call a function or not so i kept it until i determine

File: src/index.ts [hunk 2]
```diff
@@ -87,4 +88,20 @@ void (async () => {
 			error(String(e));
 		}
 	}
+
+	// If configdir is passed call than add its value to the load operation
+	if (args['confDir']) {
+		const configPath = args['confDir'];
+		try {
+			await load(configPath);
```
Comment (Creatoon) [inline src/index.ts:96]: The load function returns the config object and you are not using it anywhere.

Comment (mhmdahmedfathi) [reply src/index.ts:96]: I comment below it that the todo list didn't mention what to do with the new config object
okay what i have to do with the new config object ?

Comment (mhmdahmedfathi) [reply src/index.ts:96]: may I have to deploy package but with the config path instead of the root path?

**Comment by Creatoon** [comment]
@mhmdahmedfathi I completed confDir #61 , check this out and create another pr for serverUrl.

**Pull request closed without merging**